python/ray/worker.py

```diff
@@ -272,10 +278,22 @@ def put_object(self, value, object_id=None, pin_object=True):
                 "do this, you can wrap the ray.ObjectID in a list and "
                 "call 'put' on it (or return it).")
 
+        global should_warn_of_slow_puts
+        if should_warn_of_slow_puts:
+            start = time.perf_counter()
+
         serialized_value = self.get_serialization_context().serialize(value)
-        return self.core_worker.put_serialized_object(
+        result = self.core_worker.put_serialized_object(
             serialized_value, object_id=object_id, pin_object=pin_object)
 
+        if should_warn_of_slow_puts:
+            delta = time.perf_counter() - start
+            if delta > 0.1:
+                logger.warning("OMP_NUM_THREADS=1 is set, this may slow down "
```

Comment:
Link here would be nice too. Also, can we do this inside of `put_serialized_object`? This is also called for task arguments in `_raylet.pyx`

Reply:
Hmm I think the serialization itself may be slower.